src/arrayMethodJoin.js

```diff
@@ -5,7 +5,44 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let str = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (separator !== '' && separator !== undefined) {
+      for (let i = 0; i < this.length - 1; i++) {
+        if (this[i] === undefined || this[i] === null) {
+          this[i] = '';
+        }
+        str += `${this[i]}${separator}`;
+      }
+    }
+
+    if (separator === '') {
+      for (let i = 0; i < this.length - 1; i++) {
+        if (this[i] === undefined || this[i] === null) {
+          this[i] = '';
+        }
+        str += `${this[i]}`;
+      }
+    }
+
+    if (separator === undefined) {
+      for (let i = 0; i < this.length - 1; i++) {
+        if (this[i] === undefined || this[i] === null) {
+          this[i] = '';
+        }
+        str += `${this[i]},`;
+      }
+    }
+
+    if (this[this.length - 1]) {
+      str = str + this[this.length - 1];
+    }
```

Comment:
To much useeless check and code duplicate.
You need one loop with check `if (this[i] !== undefined && this[i] !== null) `